elastic/kibana
# [Stack Monitoring] Add stale status reporting for Kibana

## Summary

Fixes #126386

This PR adds visual warnings in the Stack Monitoring UI when one or more Kibana instances have a delay in their stats reporting. The delay can be configured with a `kibana.yml` setting and defaults to 120 seconds. 

Cluster overview:
<img width="455" alt="Screenshot 2022-06-02 at 14 18 58" src="https://user-images.githubusercontent.com/2564140/171630870-a5178e9e-2eaa-4a5d-8944-028232bc3300.png">

Kibana overview:
<img width="1333" alt="Screenshot 2022-06-02 at 14 25 44" src="https://user-images.githubusercontent.com/2564140/171630888-ab59a583-f8bd-4410-8109-721f4775f9e3.png">

Kibana instances:
<img width="1036" alt="Screenshot 2022-06-02 at 14 25 58" src="https://user-images.githubusercontent.com/2564140/171630947-906d822d-798d-435f-9de5-a4f292d556fa.png">

Kibana instances table row:

<img width="755" alt="Screenshot 2022-06-02 at 14 27 30" src="https://user-images.githubusercontent.com/2564140/171630922-d8ada44e-c962-407a-a2ee-d22057b36b02.png">

Kibana instance details:
<img width="779" alt="Screenshot 2022-06-02 at 14 36 23" src="https://user-images.githubusercontent.com/2564140/171630967-12a0b058-2e77-40d9-83f8-bcb8546cd85f.png">

### How to test

1. Set `monitoring.ui.kibana.reporting.stale_status_threshold_seconds` to something low (like 10) in your `kibana.yml`
2. Ingest some Stack Monitoring data for Kibana, either with Internal Collection or Metricbeat (easier)
3. Stop collection and wait for reports to become stale
4. Verify that the UI reports the stale status information

### To do

- [ ] Hide Stale tag in Setup mode
- [ ] Update tooltips to describe a partial state (some stale, some active)

### Checklist
